src/main/java/io/debezium/connector/vitess/transaction/RankProvider.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.vitess.transaction;
+
+import java.math.BigInteger;
+
+interface RankProvider {
+    BigInteger getRank(String transactionId);
+}
```

Comment:
Just thinking aloud; what about expanding this to allow arbitrary attributes somehow?  I'm just not sure that only a transaction Id would be sufficient.  For example, in Informix, if you don't have concurrent transactions, the underlying LSN doesn't change.  Just curious if passing a parameter object here would be more beneficial for more corner cases where a single `String` may not be sufficient.

Reply:
Responded in [Zulip](https://debezium.zulipchat.com/#narrow/stream/348255-community-vitess/topic/Increasing.20Kafka.20Partitions/near/428410735) overall we could use the source info struct to allow for all needed params to be passed to each connector.